### PR TITLE
Add ruff sort to pre-commit and sort imports in the library

### LINKED ISSUE #1252: Should we add isort to the pre-commit?
isort helps to organize the order of dependencies import, the current code is not formatted with isort. If a new PR is formatted with isort, there will be many changes in the order of dependencies import. The reordered code looks nice, but it also adds more work for the reviewers.

What about we use a PR to reformat the code, and also add isort to the pre-commit hooks?

### PR DESCRIPTION
**Pull request by glevv**
## Why are these changes needed?

To maintain clean and lean imports

## Related issue number

Closes #1252

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by glevv** [comment]
I almost sure that some tests will start to fail, so there still some work to be done

**Comment by glevv** [comment]
Should OpenAI tests also run?

**Review by thinkall — APPROVED** [review]
LGTM. Thank you very much @glevv.

**Comment by thinkall** [comment]
> Should OpenAI tests also run?

This will be triggered later when approved.

**Comment by glevv** [comment]
I'm not sure why OpenAI checks are failing.

**Comment by thinkall** [comment]
> I'm not sure why OpenAI checks are failing.

@sonichi could you take a look at this? Thanks.

File: pyproject.toml [hunk 1]
```diff
@@ -24,6 +24,7 @@ select = [
 #    "D",  # see: https://pypi.org/project/pydocstyle
 #    "N",  # see: https://pypi.org/project/pep8-naming
 #    "S",  # see: https://pypi.org/project/flake8-bandit
+    "I",  # see: https://pypi.org/project/isort/
```
Comment (Borda) [inline pyproject.toml:27]: this is the main change; the rest is just apply pre-commit :)
ref: https://github.com/microsoft/FLAML/issues/1252#issuecomment-1927245023

**Review by Borda — APPROVED** [review]

**Comment by glevv** [comment]
I don't think it is possible to automerge without all the checks being green

**Comment by Borda** [comment]
> I don't think it is possible to automerge without all the checks being green

That is right, but it seems that even all checks on mater are not green so this will need to wait until main branch is healthy =)

**Comment by Borda** [comment]
@sonichi how about adding this [bot](https://github.com/marketplace/pre-commit-ci) which would help with all this manual linitng?

**Pull request merged**